demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <utils/images_capture.h>
+#include <opencv2/gapi.hpp>
+#include <chrono>
+
+namespace custom {
+class CustomCapSource : public cv::gapi::wip::IStreamSource
+{
+public:
+    explicit CustomCapSource(std::shared_ptr<ImagesCapture>& cap, const int batch_size, const int batch_fps)
+    : cap(cap), batch_size(batch_size), batch_fps(batch_fps) {
+        prep();
+    }
+
+protected:
+    std::shared_ptr<ImagesCapture> cap;
+    int batch_size = 0; // size of batch
+    int batch_fps = 0;  // constant FPS for batch
+    cv::Mat batch;
+    int first_el = 0;   // number of first image of batch
+    int counter = 0;    // amount of images in batch
+    std::chrono::steady_clock::time_point start_time; // time from previous read()
+    bool first_pulled = false; // is first already pulled
+    void prep() {
+        GAPI_Assert(batch.empty());
+        cv::Mat frame = cap->read();
+        if (!frame.data) {
+            GAPI_Assert(false && "Couldn't grab the frame");
+        }
+        /** Create batch memory space for batch_size + 2 size
+         *  first additional element is fast image
+         *  second additional Mat is sacriface of memory for one Vec3b element
+         *  sacrifice allows avoid extra copy_n in the future **/
+        batch.create(cv::Size{frame.cols, (batch_size + 1 + 1) * frame.rows}, CV_8UC3);
+
+        /** Copy fast/real/current frame from VideoCapture to batch memory as 17th image **/
```

Comment:
> fast/real/current

Are these three different ones or it is the same frame? To avoid confusion, I'd call it with a single name throughout the code